src/include/sof/audio/dp_queue.h

```diff
@@ -25,9 +25,36 @@
  *  1) incoming and outgoing data rate MUST be the same
  *  2) Both data consumer and data producer declare max chunk sizes they want to use (IBS/OBS)
  *
- * required Buffer size:
- *	- 2*MAX(IBS,OBS) if the larger of IBS/OBS is multiplication of smaller
- *	- 3*MAX(IBS,OBS) otherwise
+ * required Buffer size is 2*MAX(IBS,OBS) to allow free read/write in various data chunk sizes
+ * and execution periods (of course in/out data rates must be same)
+ * example:
+ *  Consumer reads 3bytes each 3 cycles (IBS = 3)
+ *  producer writes 5bytes every 5 cycles (OBS = 5)
```

Comment:
in your example producer writes 3 and consumer reads 5, not as you've written. Then I think it's also more logical to start counting with 0 at cycle 0. So the first production happens after 3 (or 5) cycles

Reply:
Oops, right
I also change the example to more DP-like, consumer should start immediately when it becomes ready - meaning when it has at least 5 bytes to process

Reply:
whatever you imagine ;) may be an LL pipeline cycle - it is an abstract time unit for this example